lib/discourse_sift.rb

```diff
@@ -198,4 +126,50 @@ def self.move_to_state(post, state, opts = nil)
 
   end
 
+  # These methods are private. Do not call them directly
+  def self.reviewable_api_enabled?
+    defined?(ReviewableSiftPost)
+  end
+
+  def self.flag_post_as(post, user, topic_string)
+    # TODO: Can't get newline to render by default.  Might need to investigate overriding template or custom template?
+    # message: I18n.t('sift_flag_message') + "</br>\n" + result.topic_string
+    message = I18n.t('sift_flag_message') + topic_string
+
+    if reviewable_api_enabled?
+      PostActionCreator.create(user, post, :inappropriate, message: message)
+    else
+      post_action_type = PostActionType.types[:inappropriate]
+      PostAction.act(user, post, post_action_type, message: message)
+    end
+  rescue PostAction::AlreadyActed
+    nil # Post already flagged for this user
+  rescue Exception => e
+    Rails.logger.error("sift_debug: Exception when trying flag as system user: #{e.inspect}")
+  end
+
+  def self.remove_post_and_notify(post, reporter, reason)
+    # Post Removed Due To Content
+    PostDestroyer.new(reporter, post).destroy
+
+    # TODO: Maybe a different message if post sent to mod but still visible?
+    # Notify User
+    if SiteSetting.sift_notify_user
+      SystemMessage.create(post.user, reason, topic_title: post.topic.title)
+    end
+  end
+
+  def self.enqueue_sift_reviewable(post, reporter)
+    ReviewableSiftPost.needs_review!(
+      created_by: reporter, target: post, topic: post.topic,
+      reviewable_by_moderator: true,
+      payload: { post_cooked: post.cooked, sift: post.custom_fields[DiscourseSift::RESPONSE_CUSTOM_FIELD]  }
```

Comment:
If the Reviewable queue is running, we shouldn't be storing the repsonse in the post custom fields anymore. Everything should be in the Reviewable payload.